turbine/src/retransmit_stage.rs

```diff
@@ -444,18 +397,48 @@ impl RetransmitStage {
         rpc_subscriptions: Option<Arc<RpcSubscriptions>>,
         slot_status_notifier: Option<SlotStatusNotifier>,
     ) -> Self {
-        let retransmit_thread_handle = retransmitter(
-            retransmit_sockets,
-            quic_endpoint_sender,
-            bank_forks,
-            leader_schedule_cache,
-            cluster_info,
-            retransmit_receiver,
-            max_slots,
-            rpc_subscriptions,
-            slot_status_notifier,
+        let cluster_nodes_cache = ClusterNodesCache::<RetransmitStage>::new(
+            CLUSTER_NODES_CACHE_NUM_EPOCH_CAP,
+            CLUSTER_NODES_CACHE_TTL,
         );
 
+        let mut stats = RetransmitStats::new(Instant::now());
+
+        let mut rng = rand::thread_rng();
+        let mut shred_deduper = ShredDeduper::new(&mut rng, DEDUPER_NUM_BITS);
+
+        let thread_pool = {
+            let num_threads = get_thread_count().clamp(8, retransmit_sockets.len());
```

Comment:
The documentation says `clamp` will panic if `min > max`:
https://doc.rust-lang.org/std/cmp/trait.Ord.html#method.clamp

we probably don't want to rely on `8 <= retransmit_sockets.len()` here.

Reply:
clamp was a bad idea. I guess clippy suppressions should have a note explaining why they are there just like unsafe blocks.